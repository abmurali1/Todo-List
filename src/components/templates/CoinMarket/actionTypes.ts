export const actionTypes: {
  GET_ALL_COINS: string;
  GET_ALL_COINS_SUCCESS: string;
} = {
  GET_ALL_COINS: 'GET_ALL_COINS',
  GET_ALL_COINS_SUCCESS: 'GET_ALL_COINS_SUCCESS',
};
