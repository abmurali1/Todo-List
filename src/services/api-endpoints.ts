export function GET_ALL_COINS() {
  return `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=100&page=1&sparkline=false`;
}
export function GET_SELECT_COIN_DETAILS(coinname: string) {
  return `https://api.coingecko.com/api/v3/coins/${coinname}?localization=false&tickers=false&market_data=false&community_data=false&developer_data=false&sparkline=false `;
}
