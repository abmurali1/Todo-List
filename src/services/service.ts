import axios from 'axios';

export const request = (url: string, httpMethod: string) => {
  console.log('request--->>>', url, httpMethod);
  return new Promise(async (resolve, reject) => {
    try {
      switch (httpMethod) {
        case 'GET':
          await axios
            .get(url)
            .then(response => {
              console.log('response--->>>', response);
              if (response) {
                resolve({response: response});
              } else {
                reject(response);
              }
            })
            .catch(error => {
              console.log('error--->>>', error);
              reject(error);
            });
          break;
      }
    } catch (error) {
      console.log('error--->>>', error);
      reject(error);
    }
  });
};
