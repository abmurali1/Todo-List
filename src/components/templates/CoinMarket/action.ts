import {actionTypes} from './actionTypes';
export const getAllCoins = () => ({
  type: actionTypes.GET_ALL_COINS,
  payload: {},
});
