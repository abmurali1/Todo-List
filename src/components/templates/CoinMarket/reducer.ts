import {actionTypes} from './actionTypes';
import {AllCoinsModel} from '../../../services/models/allcoinsmodel';

type AllCoinsType = {
  allCoins: AllCoinsModel[] | null;
};

const intialState: AllCoinsType = {
  allCoins: [],
};

const AllCoinsReducer = (state = intialState, action: any) => {
  console.log('AllCoinsReducer---->>>', action);
  switch (action.type) {
    case actionTypes.GET_ALL_COINS_SUCCESS:
      return {
        ...state,
        allCoins: action.payload,
      };

    default: {
      return state;
    }
  }
};

export default AllCoinsReducer;
