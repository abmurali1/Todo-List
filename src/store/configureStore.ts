import {
  legacy_createStore as createStore,
  applyMiddleware,
  combineReducers,
} from 'redux';
import {createLogger} from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import AllCoinsReducer from '../components/templates/CoinMarket/reducer';
import {rootSaga} from '../services/sagas';
import SelectedCoinReducer from '../components/templates/CoinDetails/reducer';

const middlewares = [];
const sagaMiddleware = createSagaMiddleware();
middlewares.push(sagaMiddleware);
middlewares.push(createLogger());

export const store = createStore(
  combineReducers({AllCoinsReducer, SelectedCoinReducer}),
  applyMiddleware(...middlewares),
);
sagaMiddleware.run(rootSaga);
