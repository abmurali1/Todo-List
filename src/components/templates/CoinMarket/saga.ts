import {actionTypes} from './actionTypes';
import {GET_ALL_COINS} from '../../../services/api-endpoints';
import {put, takeLatest} from 'redux-saga/effects';
import {request} from '../../../services/service';

function* getAllCoinsWorker(action: any) {
  try {
    const {response} = yield request(GET_ALL_COINS(), 'GET');
    console.log('getAllCoinsWorker--->>>', response);
    yield put({
      type: actionTypes.GET_ALL_COINS_SUCCESS,
      payload: response.data,
    });
    //action.payload.onSuccess(response.data);
  } catch (error) {
    //action.payload.onError(error);
  }
}
export function* getAllCoinsWatcher() {
  yield takeLatest(actionTypes.GET_ALL_COINS, getAllCoinsWorker);
}
