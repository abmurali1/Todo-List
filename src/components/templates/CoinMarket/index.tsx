import React, {FC, useEffect} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  Text,
  Image,
  View,
  FlatList,
  TouchableOpacity,
} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import Header from '../../atoms/Header';

import {getAllCoins} from './action';
import I18n from '../../../localization/index';

interface CoinMarketProps {
  title: string;
  getAllCoins: Function;
  navigation: Object;
}
const CoinMarket: FC<CoinMarketProps> = props => {
  const dispatch = useDispatch();
  const coinsInfo = useSelector((state: any) => state.AllCoinsReducer.allCoins);

  useEffect(() => {
    dispatch(getAllCoins());
  }, []);

  const renderDataItem = (item: any, index: number) => {
    return (
      <View style={styles.container}>
        <Image
          source={{
            uri: item.item.image,
          }}
          style={{height: 50, width: 50}}
        />
        <Text style={{width: 70}}>{item.item.name}</Text>
        <Text>{item.item.market_cap_rank}</Text>
        <Text>{item.item.symbol}</Text>
        <Text style={{fontWeight: 'bold'}}>
          {item.item.current_price.toLocaleString('en-US', {
            style: 'currency',
            currency: 'USD',
          })}{' '}
        </Text>
        <Text>{item.item.market_cap}</Text>

        <TouchableOpacity
          onPress={() => {
            props.navigation.navigate('CoinDetails', {coinname: item.item.id});
          }}>
          <Image
            source={require('../../../assets/images/arrowRight/arrow-right.png')}
            style={{height: 20, width: 20}}
          />
        </TouchableOpacity>
      </View>
    );
  };
  return (
    <SafeAreaView style={{backgroundColor: 'lightgreen', flex: 1}}>
      <Header
        headerTitle={I18n.t('coin_market')}
        onPress={() => props.navigation.goBack()}
      />
      <View>
        <FlatList
          data={coinsInfo}
          renderItem={(item: any, index: number) => {
            return renderDataItem(item, index);
          }}
          keyExtractor={index => index.toString()}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 5,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginTop: 10,
    marginHorizontal: 5,
    borderColor: 'black',
    borderWidth: 1,
    padding: 5,
    borderRadius: 5,
  },
});

export default CoinMarket;
