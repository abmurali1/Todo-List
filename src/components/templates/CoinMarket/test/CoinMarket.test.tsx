import React from 'react';
import renderer from 'react-test-renderer';
import axios from 'axios';
import CoinMarket from '../index';
import {Provider} from 'react-redux';
import {store} from '../../../../store/configureStore';
import {GET_ALL_COINS} from '../../../../services/api-endpoints';
import {request} from '../../../../services/service';

const mockNavigate = jest.fn();
const mockSetOptions = jest.fn();

jest.mock('@react-navigation/native-stack', () => ({
  ...jest.requireActual('@react-navigation/native-stack'),
  useNavigation: () => ({
    navigate: mockNavigate,
    setOptions: mockSetOptions,
  }),
  createNavigationContainerRef: jest.fn(),
}));

jest.mock('react-native-localize', () => {
  return {
    getLocales: jest.fn(),
  };
});

jest.mock('axios', () => {
  return {
    get: jest.fn(),
  };
});

describe('get selected coins', () => {
  it('return coins', () => {
    const mockResp = {dummyData: 'Dummy data'};
    const getCoins = request(GET_ALL_COINS(), 'GET');

    expect(getCoins).resolves.toEqual(mockResp);

    expect(axios.get).toHaveBeenCalledWith(GET_ALL_COINS());

    expect(axios.get).toHaveBeenCalledTimes(1);
  });
});
