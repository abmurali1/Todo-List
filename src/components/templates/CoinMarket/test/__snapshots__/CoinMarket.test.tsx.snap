// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoinMarket> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "lightgreen",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "black",
        "borderBottomWidth": 0.2,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/arrowRight/arrow-right.png",
          }
        }
        style={
          Object {
            "height": 20,
            "marginRight": 100,
            "transform": Array [
              Object {
                "rotate": "180deg",
              },
            ],
            "width": 20,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#1b1f16",
          "fontFamily": "Verdana",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Coin Market
    </Text>
  </View>
  <View>
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
